sdk/dotnet/Pulumi/Serialization/OutputCompletionSource.cs

```diff
@@ -45,7 +45,7 @@ public void SetStringValue(string value, bool isKnown)
 
         public void SetValue(OutputData<object?> data)
             => _taskCompletionSource.SetResult(new OutputData<T>(
-                _resources.Union(data.Resources), (T)data.Value!, data.IsKnown, data.IsSecret));
+                _resources.Union(data.Resources), (data.Value == null ? default(T) : (T)data.Value)! , data.IsKnown, data.IsSecret));
```

Comment:
What exactly caused NRE in the old code? I'd guess it's `data` that would have to be `null` for NRE, which means the fix won't help?

Reply:
Nope, data.Value was null and T was a value type. `(int)null` will throw an NRE rather than returning `default(int)`.

Reply:
Oh, interesting. Thank you for the explanation!